subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/install/DefaultJavaToolchainProvisioningService.java

```diff
@@ -85,21 +86,33 @@ public DefaultJavaToolchainProvisioningService(
         this.buildPlatform = buildPlatform;
     }
 
-    public Optional<File> tryInstall(JavaToolchainSpec spec) {
+    @Override
+    public void notifyAboutActivity() {
+        if (isAutoDownloadEnabled() && toolchainResolverRegistry.requestedRepositories().isEmpty()) {
+            //todo: warn user about
+            //todo: write test for this functionality
+        }
+    }
+
+    public File tryInstall(JavaToolchainSpec spec) {
         if (!isAutoDownloadEnabled()) {
-            return Optional.empty();
+            throw new ToolchainDownloadFailedException("No locally installed toolchains match and toolchain auto-provisioning is not enabled.");
```

Comment:
Let's use the `Documentation` type to have a link to docs

Reply:
Ok, will do so.

Reply:
I guess my comment above about making the messages actionable is related to this.

AFAIU, the two possible actions are
* installing locally https://docs.gradle.org/8.0-rc-1/userguide/toolchains.html#sec:auto_detection
* adding provisioning https://docs.gradle.org/8.0-rc-1/userguide/toolchains.html#sec:provisioning

BTW I think it is really easy to miss that the simplest solution is to just use the Disco Toolchains plugin in the doc at https://docs.gradle.org/8.0-rc-1/userguide/toolchains.html#sub:download_repositories because it's a quick sentence right before explaining how to add and configure fictitious resolvers. This is confusing. I think we should make using Disco Toolchains plugin more prominent in the doc.